packages/core/src/mutants/mutant-test-planner.ts

```diff
@@ -0,0 +1,211 @@
+        return {
+          plan: PlanKind.Run,
+          mutant: {
+            ...mutant,
+            coveredBy,
+            static: isStatic,
+          },
+          runOptions: this.createMutantRunOptions(mutant, this.timeSpentAllTests, undefined),
+        };
+      }
+    } else {
+      // No coverage information exists, all tests need to run
+      return {
+        plan: PlanKind.Run,
+        mutant: {
+          ...mutant,
+          coveredBy: undefined,
+          static: undefined,
+        },
+        runOptions: this.createMutantRunOptions(mutant, this.timeSpentAllTests, undefined),
+      };
+    }
+  }
+
+  private createMutantRunOptions(mutant: MutantTestCoverage, estimatedNetTime: number, testFilter: string[] | undefined): MutantRunOptions {
+    const { disableBail, timeoutMS, timeoutFactor } = this.options;
+    const timeout = timeoutFactor * estimatedNetTime + timeoutMS + this.timeOverheadMS;
+    const hitCount = this.hitsByMutantId.get(mutant.id);
+    const hitLimit = hitCount === undefined ? undefined : hitCount * HIT_LIMIT_FACTOR;
+    return {
+      // Copy over relevant mutant fields, we don't want to copy over "static" and "coveredBy", test runners should only care about the testFilter
+      activeMutant: {
+        id: mutant.id,
+        fileName: mutant.fileName,
+        location: mutant.location,
+        mutatorName: mutant.mutatorName,
+        replacement: mutant.replacement,
+      },
+      timeout,
+      testFilter,
+      sandboxFileName: this.sandbox.sandboxFileFor(mutant.fileName),
+      hitLimit,
+      disableBail,
+      reloadEnvironment: !testFilter,
+    };
+  }
+}
+function findTestsByMutant(coveragePerTest: CoveragePerTestId | undefined, allTests: TestResult[], logger: Logger) {
+  const testsByMutantId = new Map<string, Set<TestResult>>();
+  const testsById = allTests.reduce((acc, test) => acc.set(test.id, test), new Map<string, TestResult>());
+  coveragePerTest &&
+    Object.entries(coveragePerTest).forEach(([testId, mutantCoverage]) => {
+      const foundTest = testsById.get(testId);
+      if (!foundTest) {
+        logger.warn(
+          `Found test with id "${testId}" in coverage data, but not in the test results of the dry run. Not taking coverage data for this test into account.`
+        );
+        return;
+      }
+      Object.entries(mutantCoverage).forEach(([mutantId, count]) => {
+        if (count > 0) {
+          let tests = testsByMutantId.get(mutantId);
+          if (!tests) {
+            tests = new Set();
+            testsByMutantId.set(mutantId, tests);
+          }
+          tests.add(foundTest);
+        }
+      });
+    });
+  return testsByMutantId;
+}
+
+function calculateTotalTime(testResults: Iterable<TestResult>): number {
+  let total = 0;
+  for (const test of testResults) {
+    total += test.timeSpentMs;
+  }
+  return total;
+}
+
+function toTestIds(testResults: Iterable<TestResult>): string[] {
```

Comment:
Isn't this just a `map` function?

Reply:
Same as above. Note that `reduce` and `map` might be coming to Iterables in an ES specification near you: https://github.com/tc39/proposal-collection-methods#proposal